### ISSUE TITLE
nextEvents is emptied when transition happens multiple times using StateValue

### ISSUE THREAD

**Issue by audionerd** [davidkpiano/xstate#206]
Given this machine:

```javascript
import { Machine } from 'xstate'

const statechart = {
  strict: true,
  parallel: true,
  states: {
    workspace: {
      initial: 'A',
      states: {
        'A': {
          on: {
            'enter': 'B'
          }
        },
        'B': {
          on: {
            'exit': 'C'
          },
        },
        'C': {
        }
      }
    }
  }
}

const machine = Machine(statechart)
```

If I call `machine.transition` with the same event, twice in a row, I'd expect `nextEvents` to be the same array.

Given a `State` object as the first argument, `nextEvents` is `['exit']`.

```javascript
it('nextEvents after transition given State', () => {
  const stateX = machine.transition(machine.initialState, 'enter')
  expect(stateX.nextEvents).toContain('exit') // true

  const stateY = machine.transition(stateX, 'enter')
  expect(stateY.nextEvents).toContain('exit') // true

  const stateZ = machine.transition(stateY, 'enter')
  expect(stateZ.nextEvents).toContain('exit') // true
})
```

But, given a StateValue as the first argument, `nextEvents` is `[]`:

```javascript
it('nextEvents after transition given StateValue', () => {
  const stateX = machine.transition(machine.initialState.value, 'enter')
  expect(stateX.nextEvents).toContain('exit') // true

  const stateY = machine.transition(stateX.value, 'enter')
  expect(stateY.nextEvents).toContain('exit') // true

  const stateZ = machine.transition(stateY.value, 'enter')
  // this is the bug:
  expect(stateZ.nextEvents).toContain('exit') // false
})
```

**Comment by audionerd** [comment]
To better demonstrate, here's a failing test case for `test/state.test.ts`
```
    it('returns events when transitioned from StateValue', () => {
      let A = machine.transition(machine.initialState, 'TO_THREE');
      let B = machine.transition(A.value, 'TO_THREE');
      assert.deepEqual(
        B.nextEvents,
        ['P31', 'P32', 'THREE_EVENT', 'MACHINE_EVENT']
      );
    });
```

**Comment by davidkpiano** [comment]
@audionerd Can you make a PR with that failing test case?

**Comment by davidkpiano** [comment]
Fixed in 7c1b6cd and in the latest version 4.4
